cpp/src/arrow/record_batch_test.cc

```diff
@@ -1423,37 +1453,140 @@ TEST_F(TestRecordBatch, MakeStatisticsArrayMaxApproximate) {
+
+    auto schema = ::arrow::schema({field("f0", type(byte_width_1)), field("f1", int64()),
+                                   field("f2", fixed_size_binary(byte_width_2))});
+    auto batch = RecordBatch::Make(schema, f0->length(), {f0, f1, f2});
+    ASSERT_OK_AND_ASSIGN(auto statistics_array, batch->MakeStatisticsArray());
+    ASSERT_OK_AND_ASSIGN(
+        auto expected_statistics_array,
+        MakeStatisticsArray(
+            "[null, 0, 2]",
+            {{
+                 ARROW_STATISTICS_KEY_ROW_COUNT_EXACT,
+             },
+             {
+                 ARROW_STATISTICS_KEY_MAX_VALUE_APPROXIMATE,
+             },
+             {
+                 ARROW_STATISTICS_KEY_MAX_VALUE_APPROXIMATE,
+             }},
+            {{
+                 ArrayStatistics::ValueType{int64_t{3}},
+             },
+             {
+                 ArrayStatistics::ValueType{std::string(byte_width_1, 'c')},
+             },
+             {
+                 ArrayStatistics::ValueType{std::string(byte_width_2, 'c')},
+             }},
+            {null(), type(byte_width_1), fixed_size_binary(byte_width_2)}));
+
+    AssertArraysEqual(*expected_statistics_array, *statistics_array, true);
+  }
+};
 
-  ASSERT_OK_AND_ASSIGN(auto expected_statistics_array,
-                       MakeStatisticsArray("[null, 1]",
-                                           {{
-                                                ARROW_STATISTICS_KEY_ROW_COUNT_EXACT,
-                                            },
-                                            {
-                                                ARROW_STATISTICS_KEY_MAX_VALUE_EXACT,
-                                            }},
-                                           {{
-                                                ArrayStatistics::ValueType{int64_t{3}},
-                                            },
-                                            {
-                                                ArrayStatistics::ValueType{"c"},
-                                            }}));
-  AssertArraysEqual(*expected_statistics_array, *statistics_array, true);
+using BaseBinaryOrBinaryViewOrFixedSizeBinaryLikeArrowTypes =
+    ::testing::Types<BinaryType, LargeBinaryType, BinaryViewType, FixedSizeBinaryType,
+                     StringType, LargeStringType, StringViewType>;
+
+TYPED_TEST_SUITE(TestRecordBatchMakeStatisticsArrayEachStringType,
+                 BaseBinaryOrBinaryViewOrFixedSizeBinaryLikeArrowTypes);
+
+TYPED_TEST(TestRecordBatchMakeStatisticsArrayEachStringType, EachType) {
+  this->TestEachType();
 }
 
+TEST_P(TestRecordBatchMakeStatisticsArrayFixedSizeBinaryCombination,
+       FixedSizeBinaryCombination) {
+  this->TestCombination();
+}
+
+// tuple(1, 2) tests whether RecordBatch::MakeStatistics can handle cases involving
+// different fixed_size_binary types.
+// tuple(2, 2) verifies if a single union array is created to include both values.
```

Comment:
How about creating 2 separated tests for them?

The former should use separated type objects and the latter should use the same type object. It will clarify what we want to check.

It seems that we don't need to use parameterized test for this.